teos/src/config.rs

```diff
@@ -242,6 +243,117 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// checks non-default options using HashMap
+    pub fn get_non_default_options(&self) -> HashMap<String, String> {
```

Comment:
Let's make @mariocynicys happy and make this a `Vec<(String, String)>` maybe?

Reply:
nit: Also, yo haven't reordered this to match the same order as `Config` definition. Please do.